dimension/extraction.go

```diff
@@ -0,0 +1,88 @@
+package dimension
+
+import "fmt"
+
+// ----------------------------------------------------------------------------
+
+// Extract represents an extract details
+type Extract struct {
+	Dimensions            map[string]string
+	DimensionColumnOffset int
+	ImportAPIURL          string
+	InstanceID            string
+	Line                  []string
+	MaxRetries            int
+}
+
+// InvalidNumberOfColumns is returned when the number of columns is not divisible by 3
+type InvalidNumberOfColumns struct {
+	line []string
+}
+
+func (e *InvalidNumberOfColumns) Error() string {
+	return fmt.Sprintf("invalid number of columns: [%d], needs to be divisible by 2", len(e.line))
+}
+
+// ----------------------------------------------------------------------------
+
+// New returns a new Extract object for a given instance
+func New(dimensions map[string]string, dimensionColumnOffset int, importAPIURL string, instanceID string, line []string, maxRetries int) *Extract {
+	return &Extract{
+		Dimensions:            dimensions,
+		DimensionColumnOffset: dimensionColumnOffset,
+		ImportAPIURL:          importAPIURL,
+		InstanceID:            instanceID,
+		Line:                  line,
+		MaxRetries:            maxRetries,
+	}
+}
+
+// ----------------------------------------------------------------------------
+
+// Extract method retrieves dimensions and if unique calls the sendRequest method
+func (extract *Extract) Extract() (map[string]Request, error) {
+	dimensions := make(map[string]Request)
+	line := extract.Line
+
+	// dimensionColumnOffset is the number of columns that exist
+	// prior to the columns that are relevant to the dimensions
+	dimensionColumnOffset := extract.DimensionColumnOffset + 1
+
+	// dimensionColumns is the number of columns that make a
+	// unique dimension
+	dimensionColumns := 2
```

Comment:
Like the error message the dimensions are defined by two columns in v4 csv file, if there is a possibility to change the structure of the file in the future, then yes we would this configurable, otherwise constant would suffice.